cluster-autoscaler/provisioningrequest/checkcapacity/provisioningclass.go

```diff
@@ -95,130 +95,123 @@ func (o *checkCapacityProvClass) Provision(
-			}
-			capacityAvailable = false
-		} else {
-			conditions.AddOrUpdateCondition(provReq, v1.Provisioned, metav1.ConditionTrue, conditions.CapacityIsFoundReason, conditions.CapacityIsFoundMsg, metav1.Now())
-			capacityAvailable = true
-		}
+func (o *checkCapacityProvClass) isBatchEnabled() bool {
+	return o.provreqInjector != nil && o.checkCapacityProvisioningRequestMaxBatchSize > 1
+}
 
-		_, updErr := o.client.UpdateProvisioningRequest(provReq.ProvisioningRequest)
-		if updErr != nil {
-			return false, fmt.Errorf("failed to update Provisioned condition to ProvReq %s/%s, err: %v", provReq.Namespace, provReq.Name, updErr)
+func (o *checkCapacityProvClass) checkCapacityBatch(reqs []provreq.ProvisioningRequestWithPods, combinedStatus *combinedStatusSet, startTime time.Time) []*provreqwrapper.ProvisioningRequest {
+	updates := make([]*provreqwrapper.ProvisioningRequest, 0, len(reqs))
+	for _, req := range reqs {
+		// timebox checkCapacity when batch processing.
+		if o.isBatchEnabled() && time.Since(startTime) > o.checkCapacityProvisioningRequestBatchTimebox {
+			klog.Infof("Batch timebox exceeded, processed %d check capacity provisioning requests this iteration", len(updates))
+			break
 		}
 
-		return capacityAvailable, nil
-	})
-
-	if cleanupErr != nil {
-		return false, cleanupErr
-	}
+		if err := o.checkCapacity(req.Pods, req.PrWrapper, combinedStatus); err != nil {
+			klog.Errorf("error checking capacity %v", err)
+			continue
+		}
 
-	if err != nil {
-		return false, err
+		updates = append(updates, req.PrWrapper)
 	}
-
-	return capacityAvailable, nil
+	return updates
 }
 
-// shouldStopBatchProcessing returns true if the batch processing should be stopped when:
-// - Batch processing is misconfigured
-// - Upper limits of batch processing parameters are reached
-func (o *checkCapacityProvClass) shouldStopBatchProcessing(prsProcessed int, startTime time.Time) bool {
-	if prsProcessed >= o.checkCapacityProvisioningRequestMaxBatchSize {
-		return true
-	}
-
-	if o.provreqInjector == nil {
-		klog.Errorf("ProvisioningRequestPodsInjector is not set, falling back to non-batch processing")
-		return true
-	}
+// checkCapacity checks if there is capacity, updates combinedStatus and Conditions. If capacity is found, it commits to the clusterSnapshot.
+func (o *checkCapacityProvClass) checkCapacity(unschedulablePods []*apiv1.Pod, provReq *provreqwrapper.ProvisioningRequest, combinedStatus *combinedStatusSet) error {
+	o.context.ClusterSnapshot.Fork()
 
-	if o.checkCapacityProvisioningRequestMaxBatchSize <= 1 {
-		klog.Errorf("MaxBatchSize is set to %d, falling back to non-batch processing", o.checkCapacityProvisioningRequestMaxBatchSize)
-		return true
+	// Case 1: Capacity fits.
+	if scheduled, _, err := o.schedulingSimulator.TrySchedulePods(o.context.ClusterSnapshot, unschedulablePods, scheduling.ScheduleAnywhere, true); err == nil && len(scheduled) == len(unschedulablePods) {
```

Comment:
This line is very long and has a tricky `err == nil` condition. Could you refactor it to be more readable?